### PR TITLE
NodeFSStorageAdapter: refactor

### PR DESCRIPTION
**Pull request by HerbCaudill**
No changes in behavior, just some minor refactoring for clarity.

### CODE REVIEW

**Comment by HerbCaudill** [comment]
Current dependencies on/for this PR:
* main
  * **PR #215** <a href="https://app.graphite.dev/github/pr/automerge/automerge-repo/215" target="_blank">Graphite</a>  👈
    * **PR #216** <a href="https://app.graphite.dev/github/pr/automerge/automerge-repo/216" target="_blank">Graphite</a> 
      * **PR #217** <a href="https://app.graphite.dev/github/pr/automerge/automerge-repo/217" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/automerge/automerge-repo/215?utm_source=stack-comment).

File: packages/automerge-repo-storage-nodefs/src/index.ts [hunk 1]
```diff
@@ -82,59 +64,42 @@ export class NodeFSStorageAdapter extends StorageAdapter {
...
-        }
+        const data = await this.load(key)
+        return { data, key }
       })
     )
+
+    return result
   }
 
   async removeRange(keyPrefix: string[]): Promise<void> {
```
Comment (pvh) [inline packages/automerge-repo-storage-nodefs/src/index.ts:93]: we should refuse to run this without a key value; pretty sure someone accidentally passing in undefined / empty array would delete everything they have

**Comment by pvh** [comment]
Okay! Some description would be nice for this one.

File: packages/automerge-repo-storage-nodefs/src/index.ts [hunk 2]
```diff
@@ -1,28 +1,16 @@
...
-        const childPath = path.join(dirPath, entry.name)
-        return entry.isDirectory() ? walkdir(childPath) : childPath
-      })
-    )
-  )
-
 export class NodeFSStorageAdapter extends StorageAdapter {
   private baseDirectory: string
-  private cache: {
-    [key: string]: { storageKey: StorageKey; data: Uint8Array }
```
Comment (HerbCaudill) [inline packages/automerge-repo-storage-nodefs/src/index.ts:24]: no reason to store the `storageKey` in the cache as both a value and a key.

File: packages/automerge-repo-storage-nodefs/src/index.ts [hunk 3]
```diff
@@ -82,59 +64,42 @@ export class NodeFSStorageAdapter extends StorageAdapter {
...
-    try {
-      diskFiles = await walkdir(dirPath)
-    } catch (error) {
-      if (error.code === "ENOENT") {
-        // Directory not found, initialize as empty
-        diskFiles = []
-      } else {
-        throw error
-      }
-    }
```
Comment (HerbCaudill) [inline packages/automerge-repo-storage-nodefs/src/index.ts:103]: This logic has been moved to `walkdir`

File: packages/automerge-repo-storage-nodefs/src/index.ts [hunk 4]
```diff
@@ -82,59 +64,42 @@ export class NodeFSStorageAdapter extends StorageAdapter {
...
+    const diskFiles = await walkdir(dirPath)
 
     // The "keys" in the cache don't include the baseDirectory.
     // We want to de-dupe with the cached keys so we'll use getKey to normalize them.
-    const diskKeys: string[] = diskFiles
-      .flat(Infinity) // the walk function returns a nested array
```
Comment (HerbCaudill) [inline packages/automerge-repo-storage-nodefs/src/index.ts:108]: `walkdir` returns a flattened array

File: packages/automerge-repo-storage-nodefs/src/index.ts [hunk 5]
```diff
@@ -82,59 +64,42 @@ export class NodeFSStorageAdapter extends StorageAdapter {
...
   async removeRange(keyPrefix: string[]): Promise<void> {
     const dirPath = this.getFilePath(keyPrefix)
-
-    // Warning: This method will recursively delete the directory and all its contents!
-    // Be absolutely sure this is what you want.
     await rimraf(dirPath)
   }
 
-  private getKey(key: StorageKey): string {
-    return path.join(...key)
```
Comment (HerbCaudill) [inline packages/automerge-repo-storage-nodefs/src/index.ts:137]: `getKey` and `cacheKey` do the same thing

File: packages/automerge-repo-storage-nodefs/src/index.ts (hunk 1 above)
Comment (HerbCaudill) [reply packages/automerge-repo-storage-nodefs/src/index.ts:93]: ok - that should be a separate PR

**Pull request merged**